examples/fibonacci/guest/src/main.rs

```diff
@@ -10,8 +10,12 @@ fn fib(n: u32) -> u32 {
 }
 
 fn main() {
-    // Read input from stdin.
-    let n: u32 = read(0);
+    // Read input from prover data.
+    let mut reader = ProverDataReader::new();
+    // postcard is an overkill for this example, but it is serde compatible
+    // and can be used for structs, arrays and other complex types.
+    let n: u32 = postcard::from_bytes(reader.next().unwrap()).unwrap();
```

Comment:
I think we should wrap all this in some `powdr_riscv_runtime::io` helper. The main use case is for users to just do
```rust
let x: T = powdr_riscv_runtime::io::read();
```

Reply:
The lower level interface must exist anyway, so we can optimize heavy cases.